geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java

```diff
@@ -1340,7 +1376,148 @@ protected void process(ClusterDistributionManager dm) {
       InternalCache cache = dm.getCache();
       if (cache != null) {
         TXManagerImpl mgr = cache.getTXMgr();
-        mgr.removeTransactions(this.txIds, false);
+        // check if transaction has been updated before remove it
+        mgr.removeExpiredClientTransactions(this.txIds);
+      }
+    }
+  }
+
+  public static class ExpireDisconnectedClientTransactionsMessage
```

Comment:
I recommend creating new classes as their own top-level class in their own .java file. We should get away from public inner classes that are more than simple structures.

Reply:
Made it a top level class.